internal/awsdata/kms.go

```diff
@@ -0,0 +1,79 @@
+package awsdata
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/manywho/awsinventory/internal/inventory"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// AssetTypeKMSKey is the value used in the AssetType field when fetching KMS keys
+	AssetTypeKMSKey string = "KMS Key"
+
+	// ServiceKMS is the key for the KMS service
+	ServiceKMS string = "kms"
+)
+
+func (d *AWSData) loadKMSKeys(region string) {
+	defer d.wg.Done()
+
+	kmsSvc := d.clients.GetKMSClient(region)
+
+	log := d.log.WithFields(logrus.Fields{
+		"region":  region,
+		"service": ServiceKMS,
+	})
+
+	log.Info("loading data")
+
+	var keys []*kms.KeyListEntry
+	done := false
+	params := &kms.ListKeysInput{}
+	for !done {
+		out, err := kmsSvc.ListKeys(params)
+
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+
+		keys = append(keys, out.Keys...)
+
+		if aws.BoolValue(out.Truncated) {
+			params.Marker = out.NextMarker
+		} else {
+			done = true
+		}
+	}
+
+	log.Info("processing data")
+
+	for _, k := range keys {
+		out, err := kmsSvc.DescribeKey(&kms.DescribeKeyInput{
+			KeyId: k.KeyId,
+		})
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+
+		d.results <- result{
+			Row: inventory.Row{
+				UniqueAssetIdentifier:     aws.StringValue(out.KeyMetadata.KeyId),
+				Virtual:                   true,
+				Public:                    false,
+				BaselineConfigurationName: aws.StringValue(out.KeyMetadata.Origin),
+				Location:                  region,
+				AssetType:                 AssetTypeKMSKey,
+				Comments:                  fmt.Sprintf("%s, %s", aws.StringValue(out.KeyMetadata.KeyManager), aws.StringValue(out.KeyMetadata.CustomerMasterKeySpec)),
```

Comment:
Might be good to add CreationDate and ValidTo to the comments as that's useful information.

Maybe shove them on new lines, like:
```
AWS, SYMMETRIC_DEFAULT
Created: some-date
Vaild To: some-date
```

Reply:
I guess it would be neater to construct this string outside of the `Row` instantiation

Reply:
ok, added both `CreationDate` and `ValidTo`. Tried to do it in a clean way, but let me know if you would prefer something different.